src/main/kotlin/org/jetbrains/teamcity/rest/api.kt

```diff
@@ -48,6 +51,8 @@ interface BuildLocator {
      */
     fun withAllBranches() : BuildLocator
 
+    fun withSinceBuild(locator: BuildLocator): BuildLocator
```

Comment:
It looks like TeamCity takes the last found build in sinceBuild query. Please try
`https://teamcity.jetbrains.com/app/rest/builds/buildType:(id:bt345),count:2,sinceBuild:(buildType:(id:bt345),status:FAILURE,count:2,finishDate:(date:20170621T00000+0100,condition:after))`

In general I believe that direct mapping of REST API is more important than providing some advanced mapping. Some special methods could be present, no problem, but it's terrible to downgrade to direct queries because something is missing.

Moreover, requesting the since build and than fetching the other one, it's two requests instead of one.